base_layer/transaction_components/src/transaction_components/wallet_output.rs

```diff
@@ -517,8 +597,207 @@ impl WalletOutput {
             )
             .await?;
         self.metadata_signature = metadata_sig;
+        self.fix_hash();
+        Ok(())
+    }
+
+    fn fix_hash(&mut self) {
+        let rp_hash = match &self.range_proof {
+            Some(rp) => rp.hash(),
+            None => FixedHash::zero(),
+        };
+        let output_hash = transaction_components::hash_output(
+            self.version,
+            &self.features,
+            &self.commitment,
+            &rp_hash,
+            &self.script,
+            &self.sender_offset_public_key,
+            &self.metadata_signature,
+            &self.covenant,
+            &self.encrypted_data,
+            self.minimum_value_promise,
+        );
+        self.output_hash = output_hash;
+    }
+
+    pub fn version(&self) -> TransactionOutputVersion {
+        self.version
+    }
+
+    pub fn value(&self) -> MicroMinotari {
+        self.value
+    }
+
+    pub async fn set_value<KM: TransactionKeyManagerInterface>(
+        &mut self,
+        value: MicroMinotari,
+        key_manager: &KM,
+    ) -> Result<(), TransactionError> {
+        self.input = OnceLock::new();
+        self.output = OnceLock::new();
+        self.value = value;
+        let commitment = key_manager
+            .get_commitment(&self.commitment_mask_key_id, &self.value.into())
+            .await?;
+        self.commitment = commitment;
+        let range_proof = if self.features.range_proof_type == RangeProofType::BulletProofPlus {
+            Some(
+                key_manager
+                    .construct_range_proof(
+                        &self.commitment_mask_key_id,
+                        self.value.into(),
+                        self.minimum_value_promise.into(),
+                    )
+                    .await?,
+            )
+        } else {
+            None
+        };
+        self.range_proof = range_proof;
+        self.fix_hash();
+        Ok(())
+    }
+
+    pub fn commitment_mask_key_id(&self) -> &TariKeyId {
+        &self.commitment_mask_key_id
+    }
+
+    pub async fn set_commitment_mask_key_id<KM: TransactionKeyManagerInterface>(
```

Comment:
We are setting commitment_mask not the key_id
```suggestion
    pub async fn set_commitment_mask<KM: TransactionKeyManagerInterface>(
```

Reply:
we are setting the commitment mask key id, the commitment mask would be a private key